### PR TITLE
Search dashboard refactoring

### PR DESCRIPTION
**Pull request by kangzj**
#### Changes proposed in this Pull Request:
Refactored the components to be more intuitive and moved pages to a folder to be more clear. The PR also ensures only necessary API call are made. For example, only pricing API is called on connection page and  upsell page.

#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
no
#### Testing instructions:
- Create a new JN site with current branch ([Special OP with JP Beta plugin](https://jurassic.ninja/specialops/))
- Ensure connection page, upsell page, dashboard page (including record meter) all work well

### CODE REVIEW

File: projects/packages/search/src/dashboard/components/dashboard/pages/connection-page.jsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
+/**
```
Comment (kangzj) [inline projects/packages/search/src/dashboard/components/dashboard/pages/connection-page.jsx:1]: Only added a conditional Loading component.

File: projects/packages/search/src/dashboard/components/dashboard/pages/dashboard-page.jsx [hunk 2]
```diff
@@ -145,23 +164,28 @@ export default function SearchDashboard() {
...
-				<RecordMeter
-					postCount={ postCount }
-					postTypeBreakdown={ postTypeBreakdown }
-					tierMaximumRecords={ tierMaximumRecords }
-					lastIndexedDate={ lastIndexedDate }
-				/>
+		<>
+			{ isPageLoading && <Loading /> }
+			{ ! isPageLoading && (
+				<div className="jp-search-dashboard-page">
```
Comment (kangzj) [inline projects/packages/search/src/dashboard/components/dashboard/pages/dashboard-page.jsx:170]: indents only

File: projects/packages/search/src/dashboard/components/dashboard/pages/upsell-page.jsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
+/**
```
Comment (kangzj) [inline projects/packages/search/src/dashboard/components/dashboard/pages/upsell-page.jsx:1]: Only added a conditional Loading component.

File: projects/packages/search/src/dashboard/components/dashboard/pages/loading.jsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
+/**
```
Comment (jsnmoon) [inline projects/packages/search/src/dashboard/components/dashboard/pages/loading.jsx:1]: Should this be outside the `pages` folder, given that it's more of a component than a full page?

**Review by jsnmoon — COMMENTED** [review]
The three pages work as expected and changes look reasonable; the comments made parsing the changes much easier.

Left a nitty comment inline.

File: projects/packages/search/src/dashboard/components/dashboard/pages/loading.jsx (hunk 4 above)
Comment (kangzj) [reply projects/packages/search/src/dashboard/components/dashboard/pages/loading.jsx]: Good point 👍 addressed in 28b81ab

**Review by jsnmoon — COMMENTED** [review]
The code changes look good, but it might be better to move the `pages` out of `src/dashboard/components/dashboard/pages` and into `src/dashboard/pages` for better visibility. What do you think?

**Comment by kangzj** [comment]
> The code changes look good, but it might be better to move the pages out of src/dashboard/components/dashboard/pages and into src/dashboard/pages for better visibility. What do you think?

Good point. Addressed in c556d59ec6

**Review by jsnmoon — APPROVED** [review]
:shipit: :shipit: :shipit:

**Pull request merged**